packages/server/lib/open_project.ts

```diff
@@ -0,0 +1,396 @@
+      // TODO I keep repeating this to get the resolved value
+      // probably better to have a single function that does this
+      const componentTestingEnabled = _.get(cfg, 'resolved.testingType.value', 'e2e') === 'component'
+
+      debug('createSpecWatch component testing enabled', componentTestingEnabled)
+
+      if (!this.specsWatcher) {
+        debug('watching integration test files: %s in %s', cfg.testFiles, cfg.integrationFolder)
+
+        this.specsWatcher = chokidar.watch(cfg.testFiles, {
+          cwd: cfg.integrationFolder,
+          ignored: cfg.ignoreTestFiles,
+          ignoreInitial: true,
+        })
+
+        this.specsWatcher.on('add', checkForSpecUpdates)
+
+        this.specsWatcher.on('unlink', checkForSpecUpdates)
+      }
+
+      if (componentTestingEnabled && !this.componentSpecsWatcher) {
+        debug('watching component test files: %s in %s', cfg.testFiles, cfg.componentFolder)
+
+        this.componentSpecsWatcher = chokidar.watch(cfg.testFiles, {
+          cwd: cfg.componentFolder,
+          ignored: cfg.ignoreTestFiles,
+          ignoreInitial: true,
+        })
+
+        this.componentSpecsWatcher.on('add', checkForSpecUpdates)
+
+        this.componentSpecsWatcher.on('unlink', checkForSpecUpdates)
+      }
+    }
+
+    // immediately check the first time around
+    return checkForSpecUpdates()
+  }
+
+  stopSpecsWatcher () {
+    debug('stop spec watcher')
+
+    if (this.specsWatcher) {
+      this.specsWatcher.close()
+      this.specsWatcher = null
+    }
+
+    if (this.componentSpecsWatcher) {
+      this.componentSpecsWatcher.close()
+      this.componentSpecsWatcher = null
+    }
+  }
+
+  closeBrowser () {
+    return browsers.close()
+  }
+
+  closeOpenProjectAndBrowsers () {
+    return this.closeBrowser()
+    .then(() => {
+      return this?.openProject && this?.openProject.close()
+    })
+    .then(() => {
+      this.reset()
+
+      return null
+    })
+  }
+
+  close () {
+    debug('closing opened project')
+
+    this.stopSpecsWatcher()
+
+    return this.closeOpenProjectAndBrowsers()
+  }
+
+  create (path, args: any = {}, options: any = {}) {
+    debug('open_project create %s', path)
+    debug('and options %o', options)
+
+    // store the currently open project
+    this.openProject = new ProjectBase({
+      projectType: args.testingType === 'component' ? 'ct' : 'e2e',
+      projectRoot: path,
+    })
+
+    _.defaults(options, {
+      onReloadBrowser: () => {
+        return this.relaunchBrowser?.()
+      },
+    })
+
+    if (!_.isUndefined(args.configFile)) {
+      options.configFile = args.configFile
+    }
+
+    debug('created project %s', path)
+    debug('with options %o', options)
+
+    this.options = _.extend({}, args.config, options, { args }, { testingType: args.testingType })
+
+    // return the project and return
+    // the config for the project instance
+    return this.openProject.getConfig(this.options)
```

Comment:
This is the critical change. Please compare it to the [previous code](https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/open_project.js#L355-L356).

For convenience, previously we would return `openProject.open`, which spins up a server, web sockets, etc etc.

```js
create () {
  // same up until the end
  // ...
  // ...
  return openProject.open({ ...options, testingType: args.testingType })
  .return(this)
}
```

Now we just return the config, which desktop GUI uses.